client/Crouch.lua

```diff
@@ -96,62 +127,346 @@ local function StartCrouch()
+                end
+            end
+
+            Wait(0)
+        end
+
+        -- If the crawling wasn't forcefully ended, then play the get up animations
+        PlayExitCrawlAnims(forceEnd)
+
+        -- Reset variabels
+        isCrawling = false
+        inAction = false
+        proneType = "onfront"
+        SetPedConfigFlag(PlayerPedId(), 48, false) -- CPED_CONFIG_FLAG_BlockWeaponSwitching
+
+        -- Unload animation dictionaries
+        RemoveAnimDict("move_crawl")
+        RemoveAnimDict("move_crawlprone2crawlfront")
+    end)
+end
+
+local function CrawlKeyPressed()
+    -- If we already are doing something, then don't continue
+    if inAction then
+        return
+    end
+
+    -- If already prone, then stop
+    if IsProne then
+        IsProne = false
+        return
+    end
+
+    -- If we are crouching we should stop that first
+    local wasCrouched = false
+    if isCrouched then
+        isCrouched = false
+        wasCrouched = true
+    end
+
+    local playerPed = PlayerPedId()
+    if not CanPlayerCrouchCrawl(playerPed) then
+        return
+    end
+    inAction = true
+    
+    -- If we are pointing then stop pointing
+    if Pointing then
+        Pointing = false
+    end
+
+    IsProne = true
+    SetPedConfigFlag(playerPed, 48, true) -- CPED_CONFIG_FLAG_BlockWeaponSwitching
+
+    -- Force leave stealth mode
+    if GetPedStealthMovement(playerPed) == 1 then
+        SetPedStealthMovement(playerPed, false, "DEFAULT_ACTION")
+        Wait(100)
+    end
+
+    -- Load animations that the crawling is going to use
+    LoadAnimDict("move_crawl")
+    LoadAnimDict("move_crawlprone2crawlfront")
+
+    if ShouldPlayerDiveToCrawl(playerPed) then
+        PlayAnimOnce(playerPed, "move_jump", "dive_start_run", -1)
+        Wait(1050)
+    elseif wasCrouched then
+        PlayAnimOnce(playerPed, "amb@world_human_sunbathe@male@front@enter", "enter", -1, 2.0, 0.3)
+        Wait(1500)
+    else
+        PlayAnimOnce(playerPed, "amb@world_human_sunbathe@male@front@enter", "enter", -1)
+        Wait(3000)
+    end
+
+    -- Set the player into the idle position (but only if we can still crawl)
+    if CanPlayerCrouchCrawl(playerPed) then
+        PlayIdleCrawlAnim(playerPed)
+    end
+    inAction = false
+
+    CrawlThread()
 end
-RegisterCommand('+crouch', function(source, args, raw) keyPressed = true; CrouchKeyPressed() end, false)
-RegisterCommand('-crouch', function(source, args, raw) keyPressed = false end, false)
-RegisterCommand('crouch', function(source, args, raw)
-    if not keyPressed then
+
+
+-- Commands & KeyMapping --
+if Config.CrouchEnabled then
+    if Config.CrouchKeybindEnabled then
+        RegisterKeyMapping('+crouch', "Crouch", "keyboard", Config.CrouchKeybind)
+        RegisterCommand('+crouch', function(source, args, raw) CrouchKeyPressed() end, false)
+        RegisterCommand('-crouch', function(source, args, raw) end, false) -- This needs to be here to prevent errors/warnings
```

Comment:
we don't needs these, wont be used
```suggestion
        RegisterCommand('+crouch', function() CrouchKeyPressed() end, false)
        RegisterCommand('-crouch', function() end, false) -- This needs to be here to prevent errors/warnings
```